### PR TITLE
Store locations using schema form geoname lookup service

### PR DESCRIPTION
**Pull request by mawaruDOLL**
I updated existing incident locations so the property names match the schema definition, so I also had to adjust a few places in the code that referenced these properties.

### CODE REVIEW

File: client/controllers/incidentReport.coffee [hunk 1]
```diff
@@ -1,10 +1,14 @@
+formatLocation = require '/imports/formatLocation.coffee'
+
 Template.incidentReport.helpers
   travelIcon: ->
     if @travelRelated
       return "fa-check"
     return "fa-times"
   formatDate: (date) ->
     return moment(date).format("MMM D, YYYY")
+  formatLocation: (location) ->
```
Comment (nathanathan) [inline client/controllers/incidentReport.coffee]: There is a global template helper for this.

File: client/controllers/incidentModal.coffee [hunk 2]
```diff
@@ -22,8 +22,6 @@ Template.incidentModal.events
            $(form.count).val("")
            $(form.incidentType).val(null).trigger("change")
            $(form.travelRelated).attr('checked', false)
-        if closeModal
```
Comment (nathanathan) [inline client/controllers/incidentModal.coffee]: Why was this deleted?

File: imports/schemas/incidentReport.coffee [hunk 3]
```diff
@@ -40,5 +37,32 @@ IncidentReportSchema = new SimpleSchema(
   status:
     type: String
     optional: true
+  "locations":
```
Comment (nathanathan) [inline imports/schemas/incidentReport.coffee]: This key is defined twice.

File: client/controllers/incidentModal.coffee (hunk 2 above)
Comment (mawaruDOLL) [inline client/controllers/incidentModal.coffee]: I was getting an error because closeModal wasn't defined. Should the modal be closed after "save" is clicked?

Comment (nathanathan) [inline client/controllers/incidentModal.coffee]: No, it looks like a left over from before we changed the modal buttons. You're right to delete it.

File: collections/incidents.coffee [hunk 4]
```diff
@@ -53,11 +53,32 @@ Meteor.methods
...
+              countryName: loc.countryName
+              id: loc.geonameId
+              latitude: loc.latitude
+              longitude: loc.longitude
+              name: loc.name
+            )
+          else
+            updatedLocations.push(loc)
+        if updatesMade
+          Incidents.update(incident._id, {$set: locations: {updatedLocations}})
```
Comment (nathanathan) [inline collections/incidents.coffee]: I think updatedLocations should be in an array.

**Comment by nathanathan** [comment]
The geonameId property is still used in some places:
```
/eidr-connect/collections/geolocations.coffee:
	18:         existingLocations.push(loc.geonameId)
	24:         if existingLocations.indexOf(location.geonameId.toString()) is -1
	28:             geonameId: location.geonameId,
	98:           geonameId: geolocation.geonameId

/eidr-connect/collections/incidents.coffee:
	65:           if loc.geonameId
	70:               id: loc.geonameId
```

**Comment by mawaruDOLL** [comment]
Does the geolocations collection need to be updated to use the schema as well even though it has been deprecated?

The references to geonameId in collections/incidents.coffee are in the part of the startup function that is updating locations in the counts collection to match the location schema so I don't think these lines need to be changed.

**Comment by nathanathan** [comment]
That code doesn't need to be updated, but since we're no longer using it could you delete it?

File: imports/schemas/incidentReport.coffee [hunk 5]
```diff
@@ -43,5 +40,30 @@ IncidentReportSchema = new SimpleSchema(
...
+    optional: true
+  "locations.$.countryName":
+    type: String
+  "locations.$.featureClass":
+    type: String
+  "locations.$.featureCode":
+    type: String
+  "locations.$.id":
+    type: String
+  "locations.$.latitude":
```
Comment (nathanathan) [inline imports/schemas/incidentReport.coffee]: latitude and longitude should be numbers

File: client/controllers/suggestedIncidents.coffee [hunk 6]
```diff
@@ -58,13 +58,17 @@ Template.suggestedIncidentsModal.onCreated ->
...
-          geonameId: loc.id
+          id: loc.id
           name: loc.name
-          displayName: loc.name
-          subdivision: loc.admin1Name
-          latitude: parseFloat(loc.latitude)
-          longitude: parseFloat(loc.longitude)
+          admin1Name: loc.admin1Name
+          admin2Name: loc.admin2Name
+          latitude: loc.latitude
```
Comment (nathanathan) [inline client/controllers/suggestedIncidents.coffee]: The parseFloat function should be kept

**Comment by nathanathan** [comment]
There is a new geonameId reference that was added here: https://github.com/ecohealthalliance/eidr-connect/blob/master/client/controllers/articles.coffee#L60

**Comment by nathanathan** [comment]
👍
